addons/survey/models/survey_user.py

```diff
@@ -289,24 +293,32 @@ def _prepare_statistics(self):
         }) for user_input in self)
 
         scored_questions = self.mapped('predefined_question_ids').filtered(
-            lambda question: question.question_type in ['simple_choice', 'multiple_choice']
+            lambda question: question.question_type in ['simple_choice', 'multiple_choice', 'numerical_box', 'date', 'datetime']
         )
 
         for question in scored_questions:
-            question_answer_correct = question.suggested_answer_ids.filtered(lambda answer: answer.is_correct)
+            if question.question_type in ['simple_choice', 'multiple_choice']:
```

Comment:
Suggestion:

```py
if question.question_type in ['simple_choice', 'multiple_choice']:
    res[user_input][user_input._choice_question_answer_correct(question)] += 1
else:
    res[user_input][user_input._regular_question_answer_correct(question)] += 1
```

And extract the logic to separate methods that will return 'correct', 'partial', 'skipped' or 'incorrect'.

Reply:
Done with slightly different names that I think are clearer and a 2 argument function for choice answers for what I hope will result in better computing performance.